astropy/nddata/compat.py

```diff
@@ -32,15 +32,15 @@ class NDDataArray(NDArithmeticMixin, NDSlicingMixin, NDIOMixin, NDData):
 
     Parameters
     -----------
-    data : `~numpy.ndarray` or `NDData`
+    data : array or `NDData`
```

Comment:
```suggestion
    data : ndarray or `NDData`
```

Reply:
'ndarray' links to https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html
while 'array' links to https://numpy.org/doc/stable/reference/generated/numpy.array.html#numpy.array

While both are technically correct, I think we want to link directly to `numpy.array`

Reply:
We haven't been careful elsewhere, which is a point on #11555 for followup.

Reply:
> 'ndarray' links to https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html
> while 'array' links to https://numpy.org/doc/stable/reference/generated/numpy.array.html#numpy.array
> 
> While both are technically correct, I think we want to link directly to `numpy.array`

Quite to the contrary I think. `data` takes an `ndarray` object, but `numpy.array` documents the method for creating one. Generally I find `ndarray` has the advantage of making relatively clear already from the name that it is the numpy implementation, while `array` (from just reading it without following the Xref) could be any array in the mathematical sense.

Reply:
I'll add that to the list of followups. array -> ndarray

Reply:
https://github.com/astropy/astropy/issues/11555#issue-857353639